### PR TITLE
Reviewer Rob - Monotonic timers

### PR DESCRIPTION
**Pull request by bossmc**
Fixes https://github.com/Metaswitch/chronos/issues/85.

The approach is that Chronos always talks in terms of offsets from the current time (the `start-time-delta` field in the JSON) this means we can translate from the monotonic clock on one Chronos instance to the monotonic clock on another Chronos with no loss of accuracy.

For back compatibility, Chronos also adds `start-time` based on it's local wall clock and handles receiving JSON without `start-time-delta` from old nodes.  Some future version might remove the old API.

### CODE REVIEW

**Comment by bossmc** [comment]
Also stress-tested with a live-load generator.

File: src/include/timer.h [hunk 1]
```diff
@@ -115,7 +115,7 @@ class Timer
   // Member variables (mostly public since this is pretty much a struct with utility
   // functions, rather than a full-blown object).
   TimerID id;
-  uint64_t start_time;
+  uint64_t start_time_mono_ms;
```
Comment (rkd-msw) [inline src/include/timer.h]: Is the `mono_ms` suffix really useful? I'm worried it just makes it less clear - it's not obvious what `mono` means.

File: src/test/test_timer.cpp [hunk 2]
```diff
@@ -1,42 +1,39 @@
-/**
- * @file test_timer.cpp
+/** @file test_timer.cpp
```
Comment (rkd-msw) [inline src/test/test_timer.cpp]: You should back this change out.

**Comment by rkd-msw** [comment]
Two high-level questions:

* should we have a comment or documentation somewhere obvious (https://github.com/Metaswitch/chronos/blob/dev/doc/api.md?), saying "Chronos always talks in terms of offsets from the current time; this means we can translate from the monotonic clock on one Chronos instance to the monotonic clock on another Chronos with no loss of accuracy."
* have you explicitly tested the case we were concerned about - Chronos starting up, storing some timers, and then NTP causing a large clock jump? I ask because #85 is 'Chronos is not robust to NTP time changes', not 'Chronos doesn't use monotonic time', so we should confirm that it really is fixed

Code looks good, save for a few leftover oddities from you reflowing all your code by accident.

**Comment by bossmc** [comment]
Ok, I've run the resync tests (which spotted that I can't do maths...), extended the API documentation and ran stress while changing the system clock forwards and backwards and checked that all the timers popped at the correct times.  I'll merge once the Jane Eyre release has been cut.

**Pull request merged**